homeassistant/components/fitbit/sensor.py

```diff
@@ -360,6 +361,12 @@ def __init__(
         self.extra = extra
         if self.extra is not None:
             self._attr_name = f"{self.extra.get('deviceVersion')} Battery"
+            self._attr_unique_id = (
+                f"{snakecase(self.extra.get('deviceVersion'))}_battery"
+            )
+        else:
+            self._attr_unique_id = snakecase(description.key).replace("/", "_")
+
```

Comment:
What is the use of `snakecase` here? Unique id is not visible for users, so there shouldn't be a need to "prettify" it.

Also, since there are not tests, can you please give examples of the resulting unique_id?
I am not convinced that your currently implementation is a valid source, which is why I suggested previously to add the "user-id".
Please check this page for acceptable sources for a unique id.
https://developers.home-assistant.io/docs/entity_registry_index/?_highlight=unique_id#example-acceptable-sources-for-a-unique-id

Reply:
Ohhhhhhhh I see, I'm learning a lot today! I thought that setting `unique_id` would also set the `entity_id`, so I misunderstood how this works. Now I see what you mean about adding a user-id here as well, since it doesn't affect the visible `entity_id`. I am currently looking into getting the user ID, but it might require an extra API call to fetch the user data